### PR TITLE
Update `dev/devicelab/**` to provide `--local-engine-host`.

### PR DESCRIPTION
**Pull request by matanlurey**
Partial work towards https://github.com/flutter/flutter/issues/132245.

I have to admit I don't totally understand what I've updated, or whether there are more integration points needed.

### CODE REVIEW

File: dev/devicelab/bin/run.dart [hunk 1]
```diff
@@ -95,10 +100,16 @@ Future<void> main(List<String> rawArgs) async {
       stderr.writeln(argParser.usage);
       exit(1);
     }
+    if (localEngineHost == null) {
+      stderr.writeln('When running in A/B test mode --local-engine-host is required.\n');
```
Comment (gaaclarke) [inline dev/devicelab/bin/run.dart:104]: Are we going to be hitting this when we land this?

**Review by gaaclarke — COMMENTED** [review]
The code change looks good to me.  I'm not certain how good the test coverage is for this code in pre-submit checks.  You may have to launch your own post-submit instances of a devicelab test to get a sanity check before landing.  The instructions for running these tests locally uses the `devicelab/bin/test_runner.dart`, not `devicelab/bin/run.dart` so I don't think you can verify it locally.  @zanderso may know better.

File: dev/devicelab/bin/run.dart (hunk 1 above)
Comment (matanlurey) [reply dev/devicelab/bin/run.dart:104]: My understanding is A/B test mode is just run locally (not on CI), so this should be fine with instructions?

(I could make it optional, but it potentially makes the code much more complex)

Comment (gaaclarke) [reply dev/devicelab/bin/run.dart:104]: Okay, I think this is good.

**Review by gaaclarke — APPROVED** [review]

**Pull request merged**